File: JavaFd/RegularExpressionMatching.java
public class RegularExpressionMatching {
    static class TestCase {
        String text;
        String pattern;
        boolean expected;

        public TestCase(String text, String pattern, boolean expected) {
            this.text = text;
            this.pattern = pattern;
            this.expected = expected;
        }
    }

    public TestCase[] getTestCases() {
        return new TestCase[]{
            // Given Example Test Cases
            new TestCase("aa", "a", false),
            new TestCase("aa", "a*", true),
            new TestCase("ab", ".*", true),
            new TestCase("aab", "c*a*b", true),
            new TestCase("mississippi", "mis*is*p*.", false),
            new TestCase("aab", "c*a*b", true),
            new TestCase("aaa", "ab*ac*a", true),
            new TestCase("bbab", "b*a*", false),
            new TestCase("aabcbcbcaccbcaabc", ".*a*aa*.*b*.c*.*a*", true),
            new TestCase("abbabaaaaaaacaa", "a*.*b.a.*c*b*a*c*", true),
            new TestCase("abcaaaaaaabaabcabac", ".*ab.a.*a*a*.*b*b*", true),
            new TestCase("cbaacacaaccbaabcb", "c*b*b*.*ac*.*bc*a*", true),
            new TestCase("cbccaababcbabac", "c*aab*.*b.b.*.*a*.", false),
            new TestCase("caccccaccbabbcb", "c*c*b*a*.*c*.a*a*a*", true),
            new TestCase("ccbbcabcbbaabaccc", "c*a*.*a*a*.*c*b*b*.", true),
            new TestCase("bacacaababbbcbc", ".*a*.*a*.aa*c*b*c", false),
            new TestCase("bcccccbaccccacaa", ".*bb*c*a*b*.*b*b*c*", true),

        };
    }

    public static void main(String[] args) {
        RegularExpressionMatching mainObj = new RegularExpressionMatching();
        TestCase[] testCases = mainObj.getTestCases();
        boolean success = true;
        int idx = -1;
        
        for (int i = 0; i < testCases.length; i++) {
            Solution obj = new Solution();
            TestCase testCase = testCases[i];
            boolean output = obj.isMatch(testCase.text, testCase.pattern);
            if (output != testCase.expected) {
                idx = i;
                success = false;
                break;
            }
        }
        System.out.println(success ? "Accepted" : "Wrong Answer : " + idx + "/" + testCases.length);
    }
}

File: CppFd/RegularExpressionMatching.cpp
#include <iostream>
#include <vector>
#include "Solution.cpp"

using namespace std;

class RegularExpressionMatching {
public:
    struct TestCase {
        string text;
        string pattern;
        bool expected;
        TestCase(string t, string p, bool e) : text(t), pattern(p), expected(e) {}
    };

    vector<TestCase> getTestCases() {
        return {
            // Given Example Test Cases
            {"aa", "a", false},
            {"aa", "a*", true},
            {"ab", ".*", true},
            {"aab", "c*a*b", true},
            {"mississippi", "mis*is*p*.", false},
            {"aab", "c*a*b", true},
            {"aaa", "ab*ac*a", true},
            {"bbab", "b*a*", false},
            {"aabcbcbcaccbcaabc", ".*a*aa*.*b*.c*.*a*", true},
            {"abbabaaaaaaacaa", "a*.*b.a.*c*b*a*c*", true},
            {"abcaaaaaaabaabcabac", ".*ab.a.*a*a*.*b*b*", true},
            {"cbaacacaaccbaabcb", "c*b*b*.*ac*.*bc*a*", true},
            {"cbccaababcbabac", "c*aab*.*b.b.*.*a*.", false},
            {"caccccaccbabbcb", "c*c*b*a*.*c*.a*a*a*", true},
            {"ccbbcabcbbaabaccc", "c*a*.*a*a*.*c*b*b*.", true},
            {"bacacaababbbcbc", ".*a*.*a*.aa*c*b*c", false},
            {"bcccccbaccccacaa", ".*bb*c*a*b*.*b*b*c*", true},
        };
    }

    void runTests() {
        vector<TestCase> testCases = getTestCases();
        for (size_t i = 0; i < testCases.size(); i++) {
            Solution solution;
            bool actualOutput = solution.isMatch(testCases[i].text, testCases[i].pattern);

            if (actualOutput != testCases[i].expected) {
                cout << "Wrong Answer : " << i << "/" << testCases.size() << endl;
                return;
            }
        }

        cout << "Accepted" << endl;
    }
};

int main() {
    RegularExpressionMatching tester;
    tester.runTests();
    return 0;
}
